bin/ivar_variants_to_vcf.py

```diff
@@ -57,537 +76,707 @@ def parse_args(args=None):
+            consensus - wether to merge only variants meeting consensus AF criteria
+        Returns:
+            processed_vcf_df: dataframe with consecutive variants merged
+        """
+
+        def include_rows(vcf_df, last_index, rows_to_merge):
+            indexes_to_merge = [
+                x for x in range(last_index, last_index + len(rows_to_merge))
+            ]
+            for index, row in zip(indexes_to_merge, rows_to_merge):
+                try:
+                    vcf_df.loc[index] = row
+                except ValueError:
+                    print(f"Invalid row found: {str(row)}. Skipped")
+            return vcf_df
+
+        clean_vcf_df = vcf_df[vcf_df["INFO"] == "TYPE=SNP"]
+        clean_vcf_df = clean_vcf_df[clean_vcf_df["FILTER"] == "PASS"].dropna()
+        consecutive_df = self.find_consecutive(clean_vcf_df)
+        if consecutive_df.empty:
+            return vcf_df
+        splitted_groups = self.split_non_consecutive(consecutive_df)
+        same_codon_consecutive = self.get_same_codon(splitted_groups)
+        split_rows_dict = self.split_by_codon(same_codon_consecutive)
+        for _, row_set in sorted(split_rows_dict.items()):
+            last_index = vcf_df.tail(1).index[0] + 1
+            vcf_df = vcf_df.drop(row_set.Index)
+            # Redundant "Index" column is generated by Itertuples in split_by_codon()
+            row_set = row_set.drop(["Index"], axis=1)
+            if self.find_duplicates(row_set):
+                rows_to_merge = self.handle_dup_rows(row_set.copy())
+            else:
+                clean_rows_list = self.merge_ref_alt(row_set.copy())
+                cleaned_ref_rows_list = self.remove_edge_ref(clean_rows_list)
+                rows_to_merge = self.create_merge_rowlist(cleaned_ref_rows_list.copy())
+            vcf_df = include_rows(vcf_df, last_index, rows_to_merge)
+        vcf_df = vcf_df[vcf_df["REF"] != vcf_df["ALT"]]
+        vcf_df = vcf_df.sort_index().reset_index(drop=True)
+        processed_vcf_df = vcf_df.drop_duplicates().sort_values("POS")
+        return processed_vcf_df
+
+    def get_vcf_header(self):
+        """Create the vcf header for VCFv4.2
+
+        Returns:
+            header: String containing all the vcf header lines separated by newline.
+        """
+        # Define VCF header
+        header_source = ["##fileformat=VCFv4.2", "##source=iVar"]
+        if self.ref_fasta:
+            header_contig = []
+            for record in SeqIO.parse(self.ref_fasta, "fasta"):
+                header_contig += [
+                    "##contig=<ID="
+                    + record.id
+                    + ",length="
+                    + str(len(record.seq))
+                    + ">"
+                ]
+
+            header_source += header_contig
+
+        header_info = [
+            '##INFO=<ID=TYPE,Number=1,Type=String,Description="Either SNP (Single Nucleotide Polymorphism), DEL (deletion) or INS (Insertion)">',
```

Comment:
should this be Number=1? It's a string right?

Reply:
I'll update all the header, its not only that one